include/core/mir/geometry/dimensions.h

```diff
@@ -118,6 +118,10 @@ inline constexpr DeltaX operator+(DeltaX lhs, DeltaX rhs) { return DeltaX(lhs.as
 inline constexpr DeltaY operator+(DeltaY lhs, DeltaY rhs) { return DeltaY(lhs.as_int() + rhs.as_int()); }
 inline constexpr DeltaX operator-(DeltaX lhs, DeltaX rhs) { return DeltaX(lhs.as_int() - rhs.as_int()); }
 inline constexpr DeltaY operator-(DeltaY lhs, DeltaY rhs) { return DeltaY(lhs.as_int() - rhs.as_int()); }
+inline DeltaX& operator+=(DeltaX& lhs, DeltaX rhs) { return lhs = DeltaX(lhs.as_int() + rhs.as_int()); }
+inline DeltaY& operator+=(DeltaY& lhs, DeltaY rhs) { return lhs = DeltaY(lhs.as_int() + rhs.as_int()); }
+inline DeltaX& operator-=(DeltaX& lhs, DeltaX rhs) { return lhs = DeltaX(lhs.as_int() - rhs.as_int()); }
+inline DeltaY& operator-=(DeltaY& lhs, DeltaY rhs) { return lhs = DeltaY(lhs.as_int() - rhs.as_int()); }
```

Comment:
Non-blocking, but couldn't you just implement these as `return lhs = lhs + rhs;`, since you've already got the above operators?